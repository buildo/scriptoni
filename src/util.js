import fs from 'fs';
import path from 'path';
import debug from 'debug';
debug.enable('scriptoni:*');


export function loadFileFromArgument(args, key, defaultPath) {
  const filePath = path.join(process.cwd(), args[key] ? args[key] : defaultPath);
  return fs.existsSync(filePath) && require(filePath);
}

export const logger = {
  metarpheus: debug('scriptoni:metarpheus'),
  metarpheusDiff: debug('scriptoni:metarpheus-diff'),
  lint: debug('scriptoni:lint'),
  lintStyle: debug('scriptoni:lint-style'),
  webpack: debug('scriptoni:webpack')
};
