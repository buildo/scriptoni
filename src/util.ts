import * as fs from "fs";
import * as path from "path";
import * as debug from "debug";
import * as minimist from "minimist";
import { Args } from "./model";
import { valueOrThrow } from "./scripts/webpack/util";

debug.enable("scriptoni:*");

type Key = Exclude<keyof Args, "_" | "wiro" | "bundleAnalyzer">;

export function loadFileFromArgument(
  args: Pick<Args, Key>,
  key: Key,
  defaultPath: string
) {
  const filePath = path.join(process.cwd(), args[key] || defaultPath);
  return fs.existsSync(filePath) && require(filePath);
}

export const logger = {
  bin: debug("scriptoni:bin"),
  metarpheus: debug("scriptoni:metarpheus"),
  metarpheusDiff: debug("scriptoni:metarpheus-diff"),
  lint: debug("scriptoni:lint"),
  lintStyle: debug("scriptoni:lint-style"),
  webpack: debug("scriptoni:webpack"),
  prettier: debug("scriptoni:prettier")
};

export const getArgs = (): Args => {
  return valueOrThrow(Args, minimist(process.argv.slice(2)) as any);
};
