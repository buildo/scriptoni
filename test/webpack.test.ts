import { resolve } from 'path';
import { promisify } from 'util';
import { readdir as _readdir, statSync } from 'fs';
import sortBy = require('lodash/sortBy');
import { runCommands, templateDir } from './utils';

const readdir = promisify(_readdir);

const stripHash = (fileName: string) => fileName.replace(/\.[\da-z]{20,}\./, '.');

jest.setTimeout(10 * 60 * 1000);

describe('webpack', () => {
  describe('build-ts', () => {
    beforeAll(() => runCommands([
      `cd ${templateDir}`,
      `${process.platform === 'win32' ? 'del' : 'rm -rf'} build-ts && mkdir build-ts`,
      'NODE_ENV=production ./node_modules/.bin/scriptoni web-build-ts -c ./config'
    ]));

    it('built files should stay the same', async () => {
      const fileNames = await readdir(resolve(templateDir, 'build'));
      expect(fileNames.map(stripHash)).toMatchSnapshot();
    });

    it('built files should not change size', async () => {
      const fileNames = await readdir(resolve(templateDir, 'build'));
      const fileSizes = fileNames.map(name => ({
        name: stripHash(name),
        size: Math.floor(statSync(resolve(templateDir, 'build', name)).size / 1000) * 1000
      }));
      expect(sortBy(fileSizes, 'name')).toMatchSnapshot();
    });
  });
});
