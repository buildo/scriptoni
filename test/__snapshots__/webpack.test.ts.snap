// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack build-ts built files should not change size 1`] = `
Array [
  Object {
    "name": "0.bundle.js",
    "size": 1000,
  },
  Object {
    "name": "0.bundle.js.gz",
    "size": 0,
  },
  Object {
    "name": "0.bundle.js.map",
    "size": 3000,
  },
  Object {
    "name": "1.bundle.js",
    "size": 7000,
  },
  Object {
    "name": "1.bundle.js.gz",
    "size": 1000,
  },
  Object {
    "name": "1.bundle.js.map",
    "size": 13000,
  },
  Object {
    "name": "bundle.js",
    "size": 638000,
  },
  Object {
    "name": "bundle.js.gz",
    "size": 164000,
  },
  Object {
    "name": "bundle.js.map",
    "size": 2933000,
  },
  Object {
    "name": "index.html",
    "size": 4000,
  },
  Object {
    "name": "style.min.css",
    "size": 29000,
  },
  Object {
    "name": "style.min.css.gz",
    "size": 5000,
  },
  Object {
    "name": "style.min.css.map",
    "size": 40000,
  },
]
`;

exports[`webpack build-ts built files should stay the same 1`] = `
Array [
  "0.bundle.js",
  "0.bundle.js.gz",
  "0.bundle.js.map",
  "1.bundle.js",
  "1.bundle.js.gz",
  "1.bundle.js.map",
  "bundle.js",
  "bundle.js.gz",
  "bundle.js.map",
  "index.html",
  "style.min.css",
  "style.min.css.gz",
  "style.min.css.map",
]
`;
